Add ContainsConstraintWrongActualTypeAnalyzer (#237)
nunit/nunit.analyzers: documentation/NUnit2025.md

# NUnit2025
## Wrong actual type used with ContainsConstraint.

| Topic    | Value
| :--      | :--
| Id       | NUnit2025
| Severity | Warning
| Enabled  | True
| Category | Assertion
| Code     | [ContainsConstraintWrongActualTypeAnalyzer](https://github.com/nunit/nunit.analyzers/blob/master/src/nunit.analyzers/ContainsConstraintWrongActualType/ContainsConstraintWrongActualTypeAnalyzer.cs)

## Description

The ContainsConstraint constraint requires actual argument to be either string, or collection of strings.

## Motivation

> Using ContainsConstraint with actual argument, which is not a string and not a collection of strings, leads to assertion error.

Reviewer: ```suggestion
Using a ContainsConstraint with an actual argument, which is neither a string nor a collection of strings, leads to an assertion error.
```
